src/main/java/org/sciborgs1155/robot/pivot/Pivot.java

```diff
@@ -0,0 +1,184 @@
+package org.sciborgs1155.robot.pivot;
+
+import static edu.wpi.first.units.Units.*;
+import static org.sciborgs1155.robot.pivot.PivotConstants.*;
+
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine.Direction;
+import monologue.Annotations.Log;
+import monologue.Logged;
+import org.sciborgs1155.lib.InputStream;
+import org.sciborgs1155.robot.Constants;
+import org.sciborgs1155.robot.Robot;
+import org.sciborgs1155.robot.pivot.PivotConstants.ClimbConstants;
+
+public class Pivot extends SubsystemBase implements AutoCloseable, Logged {
+  private final PivotIO pivot;
+  private final SysIdRoutine sysIdRoutine; // sysIdRoutineoogabooga
+
+  // pivot control
+  @Log.NT
+  private final ProfiledPIDController pid =
+      new ProfiledPIDController(
+          PivotConstants.kP,
+          PivotConstants.kI,
+          PivotConstants.kD,
+          new TrapezoidProfile.Constraints(PivotConstants.MAX_VELOCITY, PivotConstants.MAX_ACCEL));
+
+  private final ArmFeedforward ff =
+      new ArmFeedforward(PivotConstants.kS, PivotConstants.kG, PivotConstants.kV);
+
+  // visualization
+  @Log.NT final Mechanism2d measurement = new Mechanism2d(3, 4);
+  @Log.NT final Mechanism2d setpoint = new Mechanism2d(3, 4);
+
+  private final PivotVisualizer positionVisualizer =
+      new PivotVisualizer(measurement, new Color8Bit(255, 0, 0));
+  private final PivotVisualizer setpointVisualizer =
+      new PivotVisualizer(setpoint, new Color8Bit(0, 0, 255));
+
+  /** Creates a real or simulated pivot based on {@link Robot#isReal()} */
+  public static Pivot create() {
+    return Robot.isReal() ? new Pivot(new RealPivot()) : new Pivot(new SimPivot());
+  }
+
+  public Pivot(PivotIO pivot) {
+    this.pivot = pivot;
+    sysIdRoutine =
+        new SysIdRoutine(
+            new SysIdRoutine.Config(),
+            new SysIdRoutine.Mechanism(
+                v -> pivot.setVoltage(v.in(Volts)),
+                log -> {
```

Comment:
we don't need this anymore, since our pivot is using an absolute encoder plugged into the spark
this means our data will be caught by urcl